src/sql/parts/registeredServer/common/objectExplorerService.ts

```diff
@@ -65,16 +65,29 @@ export interface IObjectExplorerService {
 	onSelectionOrFocusChange: Event<void>;
 
 	getServerTreeView(): ServerTreeView;
+
+	getActiveConnections(): { connectionId: string, nodeInfo: sqlops.NodeInfo }[];
+
+	getChildren(connectionId: string, nodePath: string): Thenable<sqlops.NodeInfo[]>;
+
+	isExpanded(connectionId: string, nodePath: string): boolean;
+
+	expandNodeForConnection(connectionId: string, nodePath: string): Thenable<void>;
```

Comment:
Are we going to document these APIs now people can use them? In particular it's important to understand a) how to get a connection ID (what object would you pull this from) and b) how to build up a nodePath. Is this a standard thing? Would there be a way to leverage a method to build it for you?

Reply:
I'll be documenting all of it. In particular for a connection you might get it from a call to the connection APIs or from having an existing node (such as from `getActiveConnectionNodes`) and the next step of my work is to build an API that lets you get a node path from object metadata.